src/socket.cr

```diff
@@ -71,7 +71,7 @@ class Socket < IO
   # ```
   def connect(host : String, port : Int, connect_timeout = nil) : Nil
     Addrinfo.resolve(host, port, @family, @type, @protocol) do |addrinfo|
-      connect(addrinfo, timeout: connect_timeout) { |error| error }
+      connect(addrinfo, timeout: connect_timeout, &.itself)
```

Comment:
Particularly such instances dealing with error handling lose value. It's unclear what the block is about when it's just `&.itself`.
The explicitly named block argument help communicate the purpose of error handling. That makes it easier to understand the code.